src/caret_analyze/architecture/architecture.py

```diff
@@ -70,22 +70,22 @@ def get_callback_group(self, callback_group_name: str) -> CallbackGroupStructVal
 
     @property
     def callback_groups(self) -> Tuple[CallbackGroupStructValue, ...]:
-        return tuple(Util.flatten([_.callback_groups for _ in self.executors]))
+        return tuple(Util.flatten(_.callback_groups for _ in self.executors))
 
     @property
     def callback_group_names(self) -> Tuple[str, ...]:
         return tuple(sorted(_.callback_group_name for _ in self.callback_groups))
 
     @property
     def topic_names(self) -> Tuple[str, ...]:
-        return tuple(sorted({_.topic_name for _ in self.communications}))
+        return tuple(sorted(_.topic_name for _ in self.communications))
```

Comment:
This line is needed because `{}` means `set`.
{'topic_name', 'topic_name'} => {'topic_name'}

```suggestion
        return tuple(sorted({_.topic_name for _ in self.communications}))
```

Reply:
Thank you for pointing!
Fixed it in [90f6334].(https://github.com/tier4/CARET_analyze/pull/188/commits/90f6334f74169ef3d6e54d23a4c85e1f5c3a9a77)